eng/test-mounts.sh

```diff
@@ -2,36 +2,91 @@
 set -xeuo pipefail
 IFS=$'\n\t'
 
+print_to_workflow() {
+    if [[ -n ${GITHUB_STEP_SUMMARY+x} ]]; then
+        echo "$1" >> "$GITHUB_STEP_SUMMARY"
+    fi
+}
+
+file_to_workflow() {
+    # Read file content into an array
+    mapfile -t file_content < "$1"
+
+    # Print each line from the array
+    for line in "${file_content[@]}"; do
+        echo "$line" >> "$GITHUB_STEP_SUMMARY"
+    done
+}
+
 # We take the snap name as an input parameter
 snap="$1"
+arch=$(dpkg --print-architecture)
```

Comment:
**suggestion**: You may want to surround this with `""` just for consistency even though they are guaranteed to not contain a space.

```suggestion
arch="$(dpkg --print-architecture)"
```